vizro-core/src/vizro/models/_components/form/_form_utils.py

```diff
@@ -100,3 +101,21 @@ def set_default_marks(cls, marks, values):
     if not marks and values.get("step") is None:
         marks = None
     return marks
+
+
+def validate_date_picker_range(cls, range, values):
+    """Validates DatePicker `range` argument."""
+    # model return date range picker and value needs to be of type list
+    if range and isinstance(values["value"], list):
+        return range
+    # model returns date picker and value needs to be a date
+    if not range and isinstance(values["value"], date):
+        return range
+
+    if range and (len(values["value"]) == 1 or isinstance(values["value"], date)):
+        raise ValueError(
+            "The range value of the date picker is True, value argument needs to contain list of two date values."
+        )
+
+    if not range and isinstance(values["value"], list):
+        raise ValueError("The range value of the date picker is False, value argument needs to be of single date.")
```

Comment:
For the following configuration the exception is raised in build phase - when I open the app in the browser.
```
 vm.Filter(
            column="DATE_TIME",
            selector=vm.DatePicker(
                title="Pick a date",
                min="2023-01-01",
                value="2024-01-01",
                max="2024-07-01",
            ),
        ),
```
And this configuration should raise a pydantic exception before as far as I know (because range isn't specified and its True by default but the value isn't a list of two elements).